MediaGalleryUi/Ui/Component/Listing/Provider.php

```diff
@@ -23,25 +27,40 @@ class Provider extends SearchResult
      * @var GetAssetsKeywordsInterface
      */
     private $getAssetKeywords;
+    
+    /**
+     * @var GetAssetsByIdsInterface
+     */
+    private $getAssetsByIds;
+    
+    /**
+     * @var UsedIn
+     */
+    private $usedIn;
 
     /**
      * @param EntityFactory $entityFactory
      * @param Logger $logger
      * @param FetchStrategy $fetchStrategy
      * @param EventManager $eventManager
      * @param GetAssetsKeywordsInterface $getAssetKeywords
+     * @param GetAssetsByIdsInterface $getAssetsByIds
+     * @param UsedIn $usedIn
      * @param string $mainTable
      * @param null|string $resourceModel
      * @param null|string $identifierName
      * @param null|string $connectionName
      * @throws LocalizedException
+     * @SuppressWarnings(PHPMD.ExcessiveParameterList)
```

Comment:
@Nazar65 could we use current solution if constructor has more than 10 parameters? Whether it would be better to divide the addition of information (`relatedContent` ) to the picture ? 

For example by creating `afterPlugin` for current class. Something like this: 
```

<?php
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */
declare(strict_types=1);

namespace Magento\MediaGalleryUi\Plugin;

use Exception;
use Magento\Framework\Exception\LocalizedException;
use Magento\MediaGalleryApi\Api\GetAssetsByIdsInterface;
use Magento\MediaGalleryUi\Model\AssetDetailsProvider\UsedIn;
use Magento\MediaGalleryUi\Ui\Component\Listing\Provider;
use Psr\Log\LoggerInterface as Logger;

/**
 * Adds information about related content to image
 *
 * Class AddRelatedContentToImage
 */
class AddRelatedContentToImage
{
    /**
     * @var GetAssetsByIdsInterface
     */
    private $getAssetsByIds;

    /**
     * @var UsedIn
     */
    private $usedIn;
    /**
     * @var Logger
     */
    private $logger;

    /**
     * @param GetAssetsByIdsInterface $getAssetsByIds
     * @param UsedIn $usedIn
     * @param Logger $logger
     */
    public function __construct(
        GetAssetsByIdsInterface $getAssetsByIds,
        UsedIn $usedIn,
        Logger  $logger
    ) {
        $this->getAssetsByIds = $getAssetsByIds;
        $this->usedIn = $usedIn;
        $this->logger = $logger;
    }

    /**
     * Adds information about related content to image
     *
     * @param Provider $subject
     * @param array $data
     * @return array
     */
    public function afterGetData(Provider $subject, array $data): array
    {
        foreach ($data as $key => $asset) {
            try {
                $data[$key]['relatedContent'] = $this->getRelatedContent((int)$asset['id']);
            } catch (Exception $exception) {
                $this->logger->error($exception->getMessage());
            }
        }

        return $data;
    }

    /**
     * Returns information about related content
     *
     * @param int $assetId
     * @return array
     * @throws LocalizedException
     */
    private function getRelatedContent(int $assetId): array
    {
        $assetItem = $this->getAssetsByIds->execute([$assetId]);

        return isset($assetItem[0]) ? $this->usedIn->execute($assetItem[0]) : [];
    }
}

```

Thanks.